### PR TITLE
Add terraform-provider-distro image selector plugin

### PR DESCRIPTION
**Pull request by joejulian**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -55,6 +56,10 @@ RUN     wget https://github.com/samsung-cnct/terraform-provider-execute/releases
 RUN 	wget https://github.com/samsung-cnct/terraform-provider-coreosbox/releases/download/${TF_COREOSBOX_VERSION}/terraform-provider-coreosbox_linux_amd64.tar.gz && \
 	    tar -zxvf terraform-provider-coreosbox_linux_amd64.tar.gz && rm terraform-provider-coreosbox_linux_amd64.tar.gz && mv terraform-provider-coreosbox /usr/bin/
 
+	    # Adding Terraform Distro Image Selector addon
+RUN 	wget https://github.com/samsung-cnct/terraform-provider-distroimage/releases/download/${TF_DISTROIMAGE_VERSION}/terraform-provider-distroimage_linux_amd64.tar.gz && \
+	    tar -zxvf terraform-provider-distroimage_linux_amd64.tar.gz && rm terraform-provider-distroimage_linux_amd64.tar.gz && mv terraform-provider-distro /usr/bin/
+
```
Comment (guineveresaenger) [inline Dockerfile]: What does this do? I mean what's an addon (is that official Docker image language?) and what's a TF Distro Image Selector?

Comment (joejulian) [inline Dockerfile]: The official Dockerimage part of this is simply, "RUN".

This plugin is a plugin to terraform. It isn't yet used but can be after it's added to the k2-tools image (and cannot be used by k2 before it's added).

See https://github.com/samsung-cnct/terraform-provider-distroimage for more details about what the terraform plugin does.

Comment (joejulian) [inline Dockerfile]: You can look at everything after `RUN` as a bash script. This line uses wget to download a tar file from github, extract the single file that tar contains, and moves that file into /usr/bin.

Comment (davidewatson) [inline Dockerfile]: I know I'm late to the party here, but each of these curls should have checksums a la https://github.com/samsung-cnct/careen/blob/master/container/Dockerfile

Comment (joejulian) [inline Dockerfile]: Created issue #40

**Pull request merged**